packages/core/src/providers/DAppProvider.tsx

```diff
@@ -28,15 +29,38 @@ interface WithConfigProps {
 }
 
 function DAppProviderWithConfig({ children }: WithConfigProps) {
+  const [localMulticallAddresses, setLocalMulticallAddresses] = useState({})
   const { multicallAddresses, readOnlyChainId, readOnlyUrls } = useConfig()
-  const multicallAddressesMerged = { ...MULTICALL_ADDRESSES, ...multicallAddresses }
+  const multicallAddressesMerged = { ...MULTICALL_ADDRESSES, ...multicallAddresses, ...localMulticallAddresses }
+
+  const setupMulticall = (chainId: ChainId) => {
```

Comment:
Hi,

Great work,

May i suggest wrapping this function in 

```ts
const setupMulticall = useCallback((chainId: ChainId) => {
...
},[])
```

So that it won't be created at every render.